php/installer/Installer.class.inc

```diff
@@ -273,7 +273,8 @@ class Installer
         // This is apparently the most reliable way to figure out if
         // the requery is over http or https.. $_SERVER[REQUEST_SCHEME]
         // is not reliable.
-        $RequestScheme = isset($_SERVER['HTTPS']) && $_SERVER['HTTPS'] === 'on'
+        $RequestScheme = isset($_SERVER['HTTP_X_FORWARDED_PROTO'])
```

Comment:
Perhaps it would be better to check for both?

Reply:
It should definitely check for both. HTTP_X_FORWARDED_PROTO is only set if there's a proxy in between the server and PHP (and that proxy is setting the field).. it's not a reliable way to detect the protocol for servers that are directly served by a web server.

Reply:
I don't know if PHP provides a way to get the server port, but even if it did it wouldn't be reliable since the server could be configured to run on any port for either http or https.

I think

```
$RequestScheme = 'http';
if (old way of checking || x-forwarded-proto) {
 $RequestSchema = 'https';
}
```

should handle most configurations.